tools/c7n_azure/c7n_azure/query.py

```diff
@@ -30,8 +30,45 @@ def filter(self, resource_manager, **params):
         m = resource_manager.resource_type
         client = local_session(self.session_factory).client(
             "%s.%s" % (m.service, m.client))
-        # TODO: fixme
-        return client
+        enum_op, list_op = m.enum_spec
+        op = getattr(getattr(client, enum_op), list_op)
+        data = [self.to_dictionary(e) for e in op()]
+        return data
+
+    def to_dictionary(self, obj):
+        if isinstance(obj, dict):
```

Comment:
looks good for now, but i think there's a way we could get msrest layer to not deserialize to objects and just pass the dicts through as is, instead of converting multiple times.

Reply:
Totally agree, just a way to get more of the pipe lit up so people can work on different areas while we find the right way to do it.